### ISSUE TITLE
Pjax with mutiple submit buttion

### ISSUE THREAD

**Issue by dubeymahi48** [yiisoft/yii2#3719]
I am using multiple submit button. I want name of submitted button name. I am have this code in controller 
if (Yii::$app->request->getIsAjax() || Yii::$app->request->getIsPjax()) {
            if ($model->load(Yii::$app->request->post())) {
                echo $_POST['saveList'];}
                else{
                echo 'other';
                }
}

This is my button
 echo Html::submitButton($model->isNewRecord ? 'Save and Show List' : 'Update and Show List', ['name' => 'saveList', 'value' => '1', 'class' => $model->isNewRecord ? 'btn btn-success' : 'btn btn-primary']);

 echo  Html::submitButton($model->isNewRecord ? 'Save and Create New' : 'Update and View', ['name' => 'saveList', 'value' => '2', 'class' => $model->isNewRecord ? 'btn btn-success' : 'btn btn-primary']);

**Comment by Faryshta** [comment]
I think this should be moved to the yii2-pjax repository since it looks like a javascript issue.

can @yii-bot do it?

**Comment by Faryshta** [comment]
https://github.com/yiisoft/jquery-pjax/blob/master/jquery.pjax.js#L151 and

https://github.com/yiisoft/jquery-pjax/blob/master/jquery.pjax.js#L161

It seems this is an html5 and jquery issue. We might fix it with this http://stackoverflow.com/questions/15398355/jquery-ajax-does-not-get-submit-button-name i

**Comment by cebe** [comment]
@Faryshta the @yii-bot currently only works for extensions. also pjax issues may affect core and/or pjax repo so generally adding the label may not qualify for a move. Just open an issue there referring to this one.

**Comment by Faryshta** [comment]
https://jsfiddle.net/q0f5x6rh/ i built this fiddle. Looks like the `FormData` javascript object works fine meanwhile `jQuery.serializeArray()` ignores the clicked button.

I was searching for solutions on how to know which submit button triggered the event and all of them imply rewritting the way the `click` event is handled for the `<button>` element. http://stackoverflow.com/questions/3577469/form-onsubmit-determine-which-submit-button-was-pressed

and finally i searched for which browsers support the `FormData` javascript class https://developer.mozilla.org/en-US/docs/Web/API/FormData it seems this is an issue for internet explorer 9 at the moment.

we should tag this as "wont fix" and add on the docs how to work around this using `click` events in my opinion.

the other solution implies a rewrite for `jquery-pjax` events and even create new ones.

**Comment by SilverFire** [comment]
I didn't test it, but why does the execution come to that `serializeArray()` in case of POST form? I think it should not.
Or there s something magical with `window.FormData !== undefined`?

**Comment by Faryshta** [comment]
@SilverFire I provided a jsfiddle so you just need to run it.

Basically `FormData` is a class introduced in HTML5 which simplifies ajax handling with forms. It allows file uploads and knowing which button value triggered the submit.

So if the class `FormData` is undefined, then the best way to pass data to the ajax is with `serializeArray()` which lacks functionalities.

**Comment by SilverFire** [comment]
Now I get it, thank you for detailed explanation. I'm OK with tagging this issue as wont.
Could you update your PR https://github.com/yiisoft/yii2/pull/11316 and add a comment about this, please?

**Comment by Faryshta** [comment]
https://github.com/jquery/jquery/issues/2321 jquery won't fix it either. Looks like the only 'solution' is to drop support for IE 9 and alike.

**Comment by rackycz** [comment]
My workaround adds hidden inputs if a button with "name" and "value" is clicked.
Feel free to paste it into the main layout file (main.php)
The form only needs to have class "enableButtonValues".

```
document.querySelectorAll('form.enableButtonValues button[type="submit"][name][value]').forEach(button => {
    button.addEventListener('click', function (e) {
        const form = button.closest('form');
        const buttonName = button.getAttribute('name');
        const buttonValue = button.getAttribute('value');

        let hiddenInput = form.querySelector(`input[type="hidden"][name="` + buttonName + `"]`);

        if (hiddenInput) {
            hiddenInput.remove();
        }

        hiddenInput = document.createElement('input');
        hiddenInput.type = 'hidden';
        hiddenInput.name = buttonName;
        hiddenInput.value = buttonValue;

        form.appendChild(hiddenInput);
    });
});
```
